src/main/java/seedu/realodex/model/person/predicates/BirthdayIsInMonthPredicate.java

```diff
@@ -0,0 +1,76 @@
+package seedu.realodex.model.person.predicates;
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import seedu.realodex.commons.util.ToStringBuilder;
+import seedu.realodex.model.person.Person;
+
+/**
+ * Tests that a {@code Person}'s {@code Birthday} is in the Month given.
+ */
+public class BirthdayIsInMonthPredicate implements Predicate<Person> {
+    private final SimpleDateFormat monthFormat = new SimpleDateFormat("MMMM");
+    private final Optional<Calendar> month;
+
+    /**
+     * Creates object {@code BirthdayIsInMonthPredicate}
+     * @param monthName month to be tested
+     */
+    public BirthdayIsInMonthPredicate(String monthName) {
+        Optional<Calendar> tempMonth;
+        Date monthDate;
+        try {
+            monthDate = monthFormat.parse(monthName);
+            tempMonth = Optional.of(Calendar.getInstance());
+            tempMonth.get().setTime(monthDate);
+        } catch (java.text.ParseException e) {
+            // will not reach here because Parser checks for valid month
+            assert false;
+            tempMonth = Optional.empty();
```

Comment:
Once you assert false the whole program stops, you do not need to still assign tempMonth.

Reply:
i do ah cus month is final and must be intialised